Emby.Server.Implementations/Library/LiveStreamHelper.cs

```diff
@@ -48,20 +48,29 @@ public async Task AddMediaInfoWithProbe(MediaSourceInfo mediaSource, bool isAudi
 
             if (!string.IsNullOrEmpty(cacheKey))
             {
-                FileStream jsonStream = AsyncFile.OpenRead(cacheFilePath);
                 try
                 {
-                    mediaInfo = await JsonSerializer.DeserializeAsync<MediaInfo>(jsonStream, _jsonOptions, cancellationToken).ConfigureAwait(false);
+                    FileStream jsonStream = AsyncFile.OpenRead(cacheFilePath);
 
-                    // _logger.LogDebug("Found cached media info");
+                    try
+                    {
+                        mediaInfo = await JsonSerializer.DeserializeAsync<MediaInfo>(jsonStream, _jsonOptions, cancellationToken).ConfigureAwait(false);
+                        // _logger.LogDebug("Found cached media info");
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Error deserializing mediainfo cache");
+                    }
+
+                    await jsonStream.DisposeAsync().ConfigureAwait(false);
```

Comment:
```suggestion
                    await using (jsonStream.ConfigureAwait(false))
                    {
                        mediaInfo = await JsonSerializer.DeserializeAsync<MediaInfo>(jsonStream, _jsonOptions, cancellationToken).ConfigureAwait(false);
                        // _logger.LogDebug("Found cached media info");
                    }
```
IMO converting the nested try-catch into a using statement makes more sense. I believe the new disposable analyzer would flag this, so we'd need to do it eventually anyways

Reply:
Ok, just wasn't sure if the `DisposeAsync` line was needed but i assume the `using` will dispose of it even if this fails. This does look much nicer than nested try catch. Ive updated PR.